realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -196,6 +197,7 @@ private SharedRealm(OsRealmConfig osRealmConfig) {
         this.schemaInfo = new OsSchemaInfo(nativeGetSchemaInfo(nativePtr), this);
         this.context = osRealmConfig.getContext();
         this.context.addReference(this);
+        this.isPhantomRef = true;
```

Comment:
Would `isManagedByPhantomRef` be more clear? Right now it feels a bit unclear to me